### ISSUE TITLE
Error: "Session does not exist in a console request" when running feeds via a cronjob

### ISSUE THREAD

**Issue by thijskaspers** [craftcms/feed-me#863]
### Description

We are seeing errors when running Feedme from a console request. We use the wget console command via a cronjob as found in the Feedme documentation, for example: `/usr/bin/wget -O - -q -t 1 "http://your.domain/actions/feed-me/feeds/run-task?direct=1&feedId=1&passkey=FwafY5kg3c"` and while it seems to trigger and run Feedme imports perfectly, the imports show errors on entries every now and then (not all entries weirdly). For example, when importing 50 entries, about 10-15 of them seem to fail with the following error: `Session does not exist in a console request. Application.php: 129`

![111802101-ea915b80-88cd-11eb-9691-3808a3c60a10](https://user-images.githubusercontent.com/1608184/125480500-26ebb96c-86a3-4320-a53c-e97f6c39cbb1.png)

We believe it has something to do with Craft trying to get the temporary User Upload folder. While this is working fine when running a Feedme task from the control panel, it will fail once ran from the command line as there is no user session.

Until this is resolved we can't use cronjobs with Feedme, so we have to either manually trigger the feeds daily or build a custom plugin to process feeds without Feedme. Any help in fixing this is welcome :) is it a configuration issue on our end, or is this a Feedme/Craft bug?

### Additional info

- Craft version: 3.6.16
- PHP version: 7.4
- Plugins & versions: Feedme 4.3.6

**Comment by phpsa** [comment]
Fixed in Craft CMS - will be opening a pull request there for the changes:

in method: `public function getUserTemporaryUploadFolder(User $user = null)` of `vendor/craftcms/cms/src/services/Assets.php` update lines 1021 if else statment to:
```php
        if ($user) {
            $folderName = 'user_' . $user->id;
        }elseif(Craft::$app->getRequest()->getIsConsoleRequest() ){
            $folderName = 'temp_' . sha1(time());

        } else {
            // A little obfuscation never hurt anyone
            $folderName = 'user_' . sha1(Craft::$app->getSession()->id);
        }
```

**Comment by phpsa** [comment]
@thijskaspers - can you update your craft to 3.7.4 and confirm if your issue is resolved aswell.

**Comment by angrybrad** [comment]
This is resolved in Craft 3.7.4.
